fiftyone/core/fields.py

```diff
@@ -152,7 +152,9 @@ def matches_constraints(
         ):
             return False
 
-    if read_only is not None and read_only != field.read_only:
+    if read_only is not None and read_only != getattr(
```

Comment:
This method documents that it only supports `fo.Field` instances, and other stuff in here like `field.info` only supports `fo.Field`, so let's not make this change here.

We'll just confuse our future selves with seemingly unnecessary `getattr`.

Reply:
I don't think it's that confusing but sure I'll revert my commit and we can get this in